### ISSUE TITLE
Testing after last training epoch

### ISSUE THREAD

**Issue by MiguelFreire** [ultralytics/yolov5#4241]
## ❔Question
Hello, I've used a previous version of this repo, and before, the training script would perform testing on a given test set after the last training epoch. Was this removed? Or is there any argument to enable it? Couldn't find any.

Cheers,
Miguel

**Comment by glenn-jocher** [comment]
@MiguelFreire val.py runs automatically on your val set after every training epoch.

**Comment by MiguelFreire** [comment]
@glenn-jocher yeah and that is happening but previously it also had a testing step after training.
This is my data.yaml
```yaml
train: ds/train/images
val: ds/valid/images
test: ds/test/images

nc: 2
names: ['c1', 'c2']
```
A few months ago when I was using this repo a testing step was executed every time after training with the test set from data.yaml

**Comment by glenn-jocher** [comment]
@MiguelFreire code changes daily. val.py runs after every training epoch unless --noval argument is passed to train.py.

You can point val.py to any of your data splits with the --task argument:
```
python val.py --task val
python val.py --task test
```

**Comment by MiguelFreire** [comment]
Gotcha! Gonna try it! Thanks for the quick answer @glenn-jocher :)
